source/fundamentals/connection/connection-options.txt

```diff
@@ -0,0 +1,414 @@
+       |
+       | **Data Type**: IEnumerable<`MongoServerAddress <{+api-root+}/T_MongoDB_Driver_MongoServerAddress.htm>`__>
+       | **Default**: ``localhost:27017``
+     
+   * - **ServerSelectionTimeout**
+     - ``serverSelectionTimeoutMS=15000``
+     - | The length of time the driver tries to select a server before timing out.
+       |
+       | **Data Type**: ``TimeSpan``
+       | **Default**: 30 seconds
+
+   * - **SocketTimeout**
+     - ``socketTimeoutMS=0``
+     - | The length of time the driver tries to send or receive on a socket before 
+         timing out. Specifying ``0`` means no timeout.
+       |
+       | **Data Type**: ``TimeSpan``
+       | **Default**: OS default
+
+   * - **SrvMaxHosts**
+     - ``srvMaxHosts=3``
+     - | The greatest number of SRV results to randomly select when initially populating 
+         the seedlist or, during SRV polling, adding new hosts to the topology. 
+         Specifying ``0`` means no maximum.
+       |
+       | **Data Type**: {+int-data-type+}
+       | **Default**: ``0``
+
+   * - **SslSettings**
+     - ``tlsDisableCertificateRevocationCheck=false``
+     - | TLS/SSL options, including client certificates, revocation handling, and 
+         enabled and disabled TLS/SSL protocols.
+       |
+       | **Data Type**: `SslSettings <{+api-root+}/T_MongoDB_Driver_SslSettings.htm>`__
+       | **Default**: ``null``
+
+   * - **UseTls**
+     - ``tls=true`` or ``ssl=true``
+     - | Specifies whether to require TLS for connections to the server. If you use 
+         a scheme of ``\"mongodb+srv\"`` or specify other TLS options, 
+         this option defaults to ``true``.
+       |
+       | **Data Type**: {+bool-data-type+}
+       | **Default**: ``false``
+
+   * - **WaitQueueTimeout**
+     - ``waitQueueTimeoutMS=0``
+     - | The length of time the driver tries to check out a connection from a 
+         server's connection pool before timing out. Specifying ``0`` means no timeout.
+       |
+       | **Data Type**: ``TimeSpan``
+       | **Default**: 2 minutes
+
+   * - **WriteConcern**
+     - .. code-block:: none
+        :copyable: false
+     
+        w=majority
+        &wTimeoutMS=0
+        &journal=true
+
+     - | The default write-concern settings, including write timeout and 
+         journaling, for the client. Specifying ``0`` for the timeout means no timeout. 
+         See :ref:`write concern <wc-j>` for more information.
+       |
+       | **Data Type**: `WriteConcern <{+api-root+}/T_MongoDB_Driver_WriteConcern.htm>`__
+       | **Default**: ``WriteConcern.Acknowledged``
+
+   * - **WriteEncoding**
+     - {+not-available+}
+     - | Specifies whether UTF-8 string serialization is strict or lenient. With strict
+         encoding, the driver will throw an exception when it encounters an invalid 
+         UTF-8 byte sequence.
+       |
+       | **Data Type**: ``UTF8Encoding``
+       | **Default**: Strict encoding
```

Comment:
missed https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L1163

Reply:
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L1139

Reply:
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L1139

Reply:
it's only few I noticed after quick look. Not all of options we have in connectionString needs to be mentioned here, but maybe it makes sense to review it again?

Reply:
When I wrote this page, I included only the options that are part of the Connection String spec--hence, a few were left out. Below are the connection-string options in the driver code that I left out the first time.

- `connect` - [The corresponding MongoClientSettings property appears to be obsolete.](https://mongodb.github.io/mongo-csharp-driver/2.18/apidocs/html/P_MongoDB_Driver_Core_Configuration_ConnectionString_Connect.htm)

- `fsync` - Added. The MongoClientSettings property is read-only, but it looks like the connection-string option can set this to true/false. Is that right?

- `gssapiservicename` - How does this interact with the `authMechanismProperties` option? That option is covered as part of the `Credential` property.

- `maxlifetimems` - Added.

- `safe` - What does this option do? It appears to relate to the write concern, but I don't see how. And what's the corresponding `MongoClientSettings` property?

- `uuidrepresentation` - Added, though some things suggest `GuidRepresentation` is obsolete; is that true? Can you confirm the default is `GuidRepresentation.CSharpLegacy`?

- `waitqueuemultiple` - [Obsolete.](https://mongodb.github.io/mongo-csharp-driver/2.18/apidocs/html/P_MongoDB_Driver_Core_Configuration_ConnectionString_WaitQueueMultiple.htm)

- `waitqueuesize` - [Obsolete.](https://mongodb.github.io/mongo-csharp-driver/2.18/apidocs/html/P_MongoDB_Driver_Core_Configuration_ConnectionString_WaitQueueSize.htm)

- `zlibcompressionlevel` - I believe @JamesKovacs suggested not including zlib as a compression option. This is also why the `Compressors` property includes only snappy and zstd.

Reply:
* `connect ` - not needed
* `fsync` - it's part of writeConcern. Connection string allows specifying writeconcern options separately, but in MongoClientSettings we set full object (with all options including fsync) at once, so it's not readonly. To assign it you should do something close to: `clientSettings.WriteConcern = clientSettings.WriteConcern.With(fsync: false)`. I would say we need this since otherwise we will not have a way to set this option via connectionSring, but I'm not sure why it's not part of the uri spec..
* `gssapiservicename` - sounds like it's legacy option that is not part of the spec. The provided value actually just add a new authMechanism properties = `{ "SERVICE_NAME", value }`. I think we can ignore it.
* `safe` - I believe you can ignore it
* `uuidrepresentation` - this is only option for GuidRepresentationMode.V2. I believe we can ignore it since we're going to remove v2 mode in 3.0 driver release
* waitqueuemultiple / waitqueuesize  - I'm ok with not including it 
* `zlibCompressionLevel` - is part of the uri spec, so we really don't need this?

@JamesKovacs do you have any comments?

Reply:
Agreed on all those points. None of the noted URI options needs to be included in our docs. Most of those listed are deprecated. Even the ones not deprecated should only be used in very limited (and questionable) circumstances.

Regarding `gssapiServiceName`, see https://jira.mongodb.org/browse/DOCS-9897 for an explanation of its deprecation. Basically what Dima already said but in the form of a DOCS ticket.

Reply:
what about `zlibCompressionLevel`? I remember some discussions about zlib compression, but not sure whether it was decided to deprecate this option(compression?) at some point?

Reply:
note: see recent [ticket](https://jira.mongodb.org/browse/DOCSP-26706) for mongosh with a similar goal

Reply:
Leaving `zlib` out of docs, per [this discussion.](https://mongodb.slack.com/archives/C03RFE0V9NV/p1673477778917129)